CONTRIBUTING.md

```diff
@@ -160,6 +209,67 @@ Complete the prompts in the terminal.
 
 Make a pull request to merge the new branch into main. When the branch gets merged, the new contributor will show up in README.md!
 
+## API Implementation Guideline
+
+This guideline provides a high-level concept of what the ml5 library's API should look like, serving as a reference to keep the ml5 interface consistent and friendly.
+
+(Temporary note: This section does not fully reflect the current interface of the library, but rather proposes an ideal interface for future ml5.)
+
+(Temporary note 2: This section is work in progress.)
+
+### General Guidelines
+
+1. All ml5 functions called by the user should be defined in camel case.
+
+2. The user accesses an instance of the ml5 model by calling the factory function `ml5.<modelName>`. For example:
+
+```javascript
+let bodyPose = ml5.bodyPose("BlazePose");
+let neuralNetwork = ml5.neuralNetwork();
+```
+
+4. When a string parameter is passed into any ml5 function as a configuration setting, it should be matched case insensitively For example, the following calls produce the same result:
+
+   ```javascript
+   let bodyPose = ml5.bodyPose("BlazePose", { modelType: "Full" });
+   ```
+
+   ```javascript
+   let bodyPose = ml5.bodyPose("blazepose", { modelType: "full" });
+   ```
+
+5. When possible, a parameter should be optional and have default values in lieu of user definition.
+
+6. ml5 should throw a friendly warning when the user passes in an invalid parameter, and proceed with the default value when possible. For example:
+   ```javascript
+   let bodyPose = ml5.bodyPose({ modelType: "foo" });
+   // Console:
+   // Warning: bodyPose does not have modelType "foo", using modelType "full" instead.
+   ```
+7. An ml5 function should be able to accept parameters in any order. For example, the following calls produce the same result:
+
+   ```javascript
+   let bodyPose = ml5.bodyPose("blazepose", { modelType: "full" });
+   ```
+
+   ```javascript
+   let bodyPose = ml5.bodyPose({ modelType: "full" }, "blazepose");
+   ```
+
+8. Asynchronous ml5 functions should support the callback interface, and support the promise/async await interface as well when possible.
```

Comment:
```suggestion
8. Asynchronous ml5 functions should support the callback interface, and support the promise/async await interface as well when possible.
```

Are we supporting promise/async/await with the models? maybe this should say something like. "The ml5.js models do not currently support promises (async/await), however, we are considering adding this in the future and plan to follow the p5.js's lead."

Reply:
Most of the current code does support the promise interface when possible. Perhaps we can say something along the lines of "Although some existing code has both callback and promise interface, the ml5.js models do not currently support promises (async/await) officially, however, we are considering adding this in the future and plan to follow the p5.js's lead."